File: Python/BaseAdvertising.py

class BaseAdvertising:
    """base class that tracks views and clicks"""
    def __init__(self):
        self.__clicks = 0
        self.__views = 0

    def get_clicks(self):
        return self.__clicks

    def get_views(self):
        return self.__views

    def inc_clicks(self):
        self.__clicks += 1

    def inc_views(self):
        self.__views += 1

    def describe_me(self):
        return self.__doc__

File: Cpp/src/BaseAdvertising.cpp
#include <iostream>
#include "BaseAdvertising.hpp"
using namespace std;


int BaseAdvertising::get_clicks(){
    return clicks;
}

int BaseAdvertising::get_views(){
    return views;
}

void BaseAdvertising::inc_clicks(){
    clicks++;
}

void BaseAdvertising::inc_views(){
    views++;
}

string BaseAdvertising::describe_me(){
    return "Base Advertising class is a base class that tracks views and clicks";
}